### PR TITLE
Api renames

### PR DESCRIPTION
**Pull request by usmansaleem**
# PR Description

- Changed OpenAPI paths to seperate ETH1 and ETH2 operations.
`/api/v1/eth2/sign/{identifier}`
`/api/v1/eth1/sign/{identifier}`
`/api/v1/eth2/publicKeys`
`/api/v1/eth1/publicKeys`
`/upcheck`

- Following paths are removed
`/signer/sign/{identifier}`
`/signer/publicKeys/{keyType}`

- Removed OpenAPI equivalent JSON-RPC operations from `/rpc/v1`.  (Filecoin JSON-RPC operations are still available at `/rpc/v1/filecoin`)
- Update metrics
- Update relevant test cases.

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 1]
```diff
@@ -147,20 +140,16 @@ public void run() {
       final FileCoinArtifactSignerProvider fcArtifactSignerProvider =
           new FileCoinArtifactSignerProvider(fcBlsSignerProvider, fcSecpSignerProvider);
 
+      // create router from OpenApi spec - web3signer.yaml
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:143]: not sure if this is for now - but I'd love to see us split the configuration/buildup into Eth1, Eth2 and Filecoin "partitions" - thus the secp and the bls identifiers would never need to be shipped to a function together - they are fundamentally different, individual entities.

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 2]
```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
...
-    openAPI3RouterFactory.addFailureHandlerByOperationId(UPCHECK_OPERATION_ID, errorHandler);
-
-    // public key handler
-    openAPI3RouterFactory.addHandlerByOperationId(
-        GET_PUBLIC_KEYS_OPERATION_ID,
-        new BlockingHandlerDecorator(new GetPublicKeysHandler(keyIdentifiers), false));
-    openAPI3RouterFactory.addFailureHandlerByOperationId(
-        GET_PUBLIC_KEYS_OPERATION_ID, errorHandler);
+    // public key list handler
+    addEth1PublicKeysListHandler(openAPI3RouterFactory, secpIdentifiers, errorHandler);
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:181]: _personally_ prefer to think about this stuff as eth1 and eth2, rather than signing and public_keys APIs ... but that may be a personal preference in how to split the problem space.

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 3]
```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
...
+      final SignerForIdentifier<BlsArtifactSignature> blsSigner,
+      final MetricsSystem metricsSystem,
+      final LogErrorHandler errorHandler) {
     openAPI3RouterFactory.addHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID,
+        ETH2_SIGN.name(),
         new BlockingHandlerDecorator(
-            new SignForIdentifierHandler(blsSigner, metricsSystem, "bls"), true));
+            new SignForIdentifierHandler(
+                blsSigner, new OpenApiOperationsMetrics(metricsSystem, BLS)),
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:203]: what is OpenApiOperationsMetrics? And how does it work - FEELS like this might be another change for another PR?

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 4]
```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
...
+            false));
+    openAPI3RouterFactory.addFailureHandlerByOperationId(ETH2_SIGN.name(), errorHandler);
+  }
+
+  private void addEth1SignHandler(
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:208]: this feels really similar to the Eth2 function - can it be parameterised and duplication removed?

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/UpcheckHandler.java [hunk 5]
```diff
@@ -15,16 +15,12 @@
...
 import io.vertx.core.Handler;
 import io.vertx.ext.web.RoutingContext;
 
 public class UpcheckHandler implements Handler<RoutingContext> {
-  final Upcheck upcheck = new Upcheck();
-
   @Override
   public void handle(final RoutingContext routingContext) {
-    routingContext.response().putHeader(CONTENT_TYPE, TEXT_PLAIN_UTF_8).end(upcheck.status());
+    routingContext.response().putHeader(CONTENT_TYPE, TEXT_PLAIN_UTF_8).end("OK");
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/UpcheckHandler.java:24]: So do we need the upcheck class anymore? Or has it been removed?

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/metrics/OpenApiOperationsMetrics.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
+
+public class OpenApiOperationsMetrics {
+  private static Counter signersLoadedCounter;
+
+  private final Counter malformedRequestCounter;
+  private final OperationTimer signingTimer;
+  private final Counter missingSignerCounter;
+
+  public OpenApiOperationsMetrics(final MetricsSystem metricsSystem, final KeyType keyType) {
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/metrics/OpenApiOperationsMetrics.java:29]: are these actually "OpenApiOperationsMetrics" - or maybe "RouteMetrics"? Not sure ... or are they really SignOperationMetrics?

File: core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcJsonRpc.java [hunk 7]
```diff
@@ -102,7 +101,7 @@ public boolean filecoinWalletHas(@JsonRpcParam("address") final String address)
   public FilecoinSignedMessage filecoinSignMessage(
       @JsonRpcParam("identifier") final String identifier,
       @JsonRpcParam("message") final FilecoinMessage message) {
-    metrics.incwWalletSignMessageRequestCounter();
+    metrics.incWalletSignMessageRequestCounter();
```
Comment (rain-on) [inline core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/FcJsonRpc.java:104]: thanks :) for the typo fix.

File: core/src/main/resources/openapi/web3signer.yaml [hunk 8]
```diff
@@ -57,23 +57,79 @@ paths:
...
           in: 'path'
           required: true
-          description: 'Type of public keys to return'
+          description: 'Key for which data to sign'
           schema:
-            type: string
-            enum:
-              - BLS
-              - SECP256K1
+            type: 'string'
```
Comment (rain-on) [inline core/src/main/resources/openapi/web3signer.yaml:73]: might need to ask JFrame to take a look at this bit,  but looks ok to me.

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java (hunk 3 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/Runner.java:203]: I simply wrapped scattered signing/http metrics into this class. Makes the runner class and SignForIdentifierHandler a bit simpler with regards to creating the matrices.

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/UpcheckHandler.java (hunk 5 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/UpcheckHandler.java:24]: The Upcheck class has been removed.

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/metrics/OpenApiOperationsMetrics.java (hunk 6 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/metrics/OpenApiOperationsMetrics.java:29]: I think all of them are Sign operation metrics except for 1 which you had categorised under "http". Suggestions for class name??

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 9]
```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
...
-        });
-    openAPI3RouterFactory.addFailureHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID, errorHandler);
+        ETH1_LIST.name(),
+        new BlockingHandlerDecorator(new PublicKeysListHandler(secpIdentifiers), false));
+    openAPI3RouterFactory.addFailureHandlerByOperationId(ETH1_LIST.name(), errorHandler);
+  }
 
-    return openAPI3RouterFactory;
+  private void addEth2PublicKeysListHandler(
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:232]: similar to the above comment, looks like some dupe here, these publicKeysListHandlers are exactly the same except for some  naming

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/Runner.java]: done. HttpApiMetrics makes more sense.

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java (hunk 9 above)
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/Runner.java]: refactored out to common logic.

**Review by jframe — APPROVED** [review]
